sequencer/src/block2.rs

```diff
@@ -57,31 +57,53 @@ impl BlockPayload {
         Some(Self { payload })
     }
 
+    #[allow(dead_code)] // TODO temporary
+    fn from_bytes<B>(bytes: B) -> Self
+    where
+        B: IntoIterator<Item = u8>,
+    {
+        Self {
+            payload: bytes.into_iter().collect(),
+        }
+    }
+
     // Return the range `r` such that the `index`th tx bytes are at `self.payload[r]`.
-    fn get_tx_range(&self, index: TxIndex) -> Option<Range<usize>> {
-        let tx_bodies_offset = self.tx_bodies_offset()?;
```

Comment:
Why did we get rid of the `tx_bodies_offset` function?

Reply:
Collateral damage from a chaotic dev process. Restored in 35499ad84f8fe4d19a65665aa985330c64567eaa